fix: support tuple iterator cache in integration tests

senojj opened the pull request:
<!-- Thanks for opening a PR! Here are some quick tips:
If this is your first time contributing, [read our Contributing Guidelines](https://github.com/openfga/.github/blob/main/CONTRIBUTING.md) to learn how to create an acceptable PR for this repo.
By submitting a PR to this repository, you agree to the terms within the [OpenFGA Code of Conduct](https://github.com/openfga/.github/blob/main/CODE_OF_CONDUCT.md)

If your PR is under active development, please submit it as a "draft". Once it's ready, open it up for review.
-->

<!-- Provide a brief summary of the changes -->

## Description
<!-- Provide a detailed description of the changes -->
#### What problem is being solved?
Currently, the tuple iterator cache TTL is set to 1 nanosecond when running integration tests. This is necessary because 3 tests cannot currently work with the cache in play:
- ttu_discard_invalid
- userset_discard_invalid
- userset_discard_invalid_wildcard

However, this effectively disables the cache for testing purposes and adds the potential for these tests to be flaky. These tests need not be designed in such a way.
#### How is it being solved?
The tests are changed so that their subsequent stage assertions do not depend upon or conflict with the tuples of the previous stages.

#### What changes are made to solve it?
The three tests are updated to use different user and object identifiers from previous stages.
The tuple iterator cache TTL limitation is removed, allowing the default value to be used.

## References
<!--
Provide a list of any applicable references here (GitHub Issue, [OpenFGA RFC](https://github.com/openfga/rfcs), other PRs, etc..). We prefer an accompanying issue for all non-trivial PRs.

When referencing links, follow these examples:
* closes https://github.com/openfga/{repo}/issues/{issue_number}
* reverts https://github.com/openfga/{repo}/pull/{pr_number}
* followup https://github.com/openfga/{repo}/pull/{pr_number}
* blocked by https://github.com/openfga/{repo}/pull/{pr_number}
-->

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Tests**
  * Updated test cases with new object and user identifiers, and added new tuples to reflect updated user-role relationships.
  * Adjusted test data in permission checks to use the new identifiers and relationships.
  * Removed explicit cache TTL configuration from test setup, now using default cache settings.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

adriantam reviewed (commented):
The original intend of the tests was to assert whether check returns valid value when the model is changed.  I agree that we should remove the small wait as it does not produce stable value. However, since these tests become invalid due to introduction of caches, we should choose either of the two routes:
- remove the cache in these tests
- remove these "discard" tests.

senojj commented:
I've changed the tests to make them useful. Their original state when I approached them was not entirely useful because they didn't actually assert that the obsolete tuple was ignored. This is because there were no relationships to the obsolete tuple. I've addressed this issue and have included assertions that confirm that the obsolete tuple is ignored.
